### ISSUE TITLE
Remote page attachments broken on iOS webviews

### ISSUE THREAD

**Issue by gmajoulet** [ampproject/amphtml#33709]
Remote page attachments are broken on iOS webviews, or at least in the Google native app and Discover.

All history navigations need to originate from a clicked anchor tag. We should replace our navigation based on the history API, and trigger a click on the anchor tag.

https://github.com/ampproject/amphtml/blob/e03dba1b1fe7ff1d4dfeb59ba257ab4d0d93341f/extensions/amp-story/1.0/amp-story-page-attachment.js#L294-L298

The swipe up interaction we get is not a trusted interaction on Safari (it is on Chrome), but because this is a `_self` navigation, it won't be blocked by the browser/webview.
This fix has been prototyped and tested already.

cc @ampproject/wg-stories

**Comment by choumx** [comment]
I assume #33656 is the candidate fix for this. Curious if this bug is a regression and why P1 vs. P0?

**Comment by gmajoulet** [comment]
It is not a regression, the technical details are complicated but the TLDR is that the proprietary viewer used in the iOS Google app doesn't handle JavaScript triggered navigation well, only clicked anchors, which is pretty hacky but does work around these limitations.
It's very specific to this one implementation, which has been broken for months. Do you want to try applying for same day shipping and file a cherry pick request?

**Comment by choumx** [comment]
Shucks I canceled Prime but let's see if we need to expedite next week.

@mszylkowski How likely is it that we'll have the fix in next week's cut?

**Comment by mszylkowski** [comment]
The fix to this is working in that PR, we need to make a P3 work with the swipe ups as well, but we could do it in another PR if we want to merge this faster

**Comment by choumx** [comment]
> we need to make a P3 work with the swipe ups as well

Can you elaborate on this? "Swipe ups" is distinct from external link page attachments?

**Comment by mszylkowski** [comment]
Yes! We were thinking to fix a P3 on this PR as well but decided to leave it for later (tracked on #33860), meaning we can merge the PR #33656 and close this issue as soon as the checks pass.

Also, external page attachments are activated with a swipe up, that's why those terms are used interchangeably

**Comment by choumx** [comment]
Ah ok, an analytics bug fix follow-up is fine for us.

**Comment by mszylkowski** [comment]
Analytics fix is #33911. Closing this bug since #33656 is merged.
